go/kvstore/boxer.go

```diff
@@ -0,0 +1,277 @@
+package kvstore
+
+import (
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/keybase/client/go/kbcrypto"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/msgpack"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-crypto/ed25519"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+type KVStoreBoxer interface {
+	Box(mctx libkb.MetaContext, entryID keybase1.KVEntryID, revision int, cleartextValue string) (ciphertext string,
+		teamKeyGen keybase1.PerTeamKeyGeneration, ciphertextVersion int, err error)
+	Unbox(mctx libkb.MetaContext, entryID keybase1.KVEntryID, revision int, ciphertext string, formatVersion int,
+		senderUID keybase1.UID, senderEldestSeqno keybase1.Seqno, senderDeviceID keybase1.DeviceID) (cleartext string, err error)
```

Comment:
I'd like to get max involved with reviewing the crypto. It could be that [signencrypt](https://github.com/keybase/client/blob/master/go/chat/signencrypt/codec.go#L229) is a tool for this job

Reply:
done!